### PR TITLE
feat(ISV-5447): add multi-arch and sha info to release note images

### PR DESCRIPTION
**Pull request by wcheang**
This is ready for review, but depends on https://github.com/konflux-ci/release-service-utils/pull/333/ to be merged first so that the multiarch field is populated before merge.

### CODE REVIEW

**Review by mmalina — DISMISSED** [review]

**Review by johnbieren — CHANGES_REQUESTED** [review]
If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json

**Comment by wcheang** [comment]
> If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json

Thanks @johnbieren, added the schema.

**Comment by wcheang** [comment]
My PTO is starting tomorrow, so @jedinym from my team will be taking over getting this PR merged.

**Comment by mmalina** [comment]
> If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json

This is added during the release pipeline. I thought the schema only covers what's there at the start, but apparently I was wrong. What's the purpose of having releaseNotes.content.images there? It will never be validated, no?

File: tasks/populate-release-notes-images/populate-release-notes-images.yaml
Comment (mmalina) [inline tasks/populate-release-notes-images/populate-release-notes-images.yaml:1]: Do not merge. The image needs to be updated with https://github.com/konflux-ci/release-service-utils/pull/333 first.

For the future, please create the PR as a Draft or at least create a thread as a reminder that something still needs to be done. Otherwise it can lead to premature merge as just happened here: https://github.com/hacbs-release/app-interface-deployments/pull/245

Comment (johnbieren) [reply tasks/populate-release-notes-images/populate-release-notes-images.yaml:1]: Moved to draft

**Comment by johnbieren** [comment]
> > If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json
> 
> This is added during the release pipeline. I thought the schema only covers what's there at the start, but apparently I was wrong. What's the purpose of having releaseNotes.content.images there? It will never be validated, no?

The releaseNotes stuff is unique. We don't overwrite. We `=+` https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/populate-release-notes-images/populate-release-notes-images.yaml#L111
A user could define the same stuff the task is adding for them. I don't know why they'd want to, but they could. So, if a field *could* be provided by the user, I think it should be in the schema

**Review by johnbieren — COMMENTED** [review]
I replied on the ticket, but why are we adding new fields to the advisories? Shouldn't this be driven by prodsec? Maybe it already is, but I haven't seen it

**Comment by mmalina** [comment]
> > > If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json
> > 
> > 
> > This is added during the release pipeline. I thought the schema only covers what's there at the start, but apparently I was wrong. What's the purpose of having releaseNotes.content.images there? It will never be validated, no?
> 
> The releaseNotes stuff is unique. We don't overwrite. We `=+` https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/populate-release-notes-images/populate-release-notes-images.yaml#L111 A user could define the same stuff the task is adding for them. I don't know why they'd want to, but they could. So, if a field _could_ be provided by the user, I think it should be in the schema

I'm not completely sure I buy this argument - it seems to me that being theoretically able to set this already in the RPA is more of a side effect. I'm not even sure if that should be allowed at all - it should say what was actually included in that release, right? So users shouldn't really be able to say that something extra is being released too when it's not. Or am I missing something?

**Comment by mmalina** [comment]
> I replied on the ticket, but why are we adding new fields to the advisories? Shouldn't this be driven by prodsec? Maybe it already is, but I haven't seen it

Good point. I completely missed the fact that this will also add these fields to the resulting advisory json.

**Comment by johnbieren** [comment]
> > > > If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json
> > > 
> > > 
> > > This is added during the release pipeline. I thought the schema only covers what's there at the start, but apparently I was wrong. What's the purpose of having releaseNotes.content.images there? It will never be validated, no?
> > 
> > 
> > The releaseNotes stuff is unique. We don't overwrite. We `=+` https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/populate-release-notes-images/populate-release-notes-images.yaml#L111 A user could define the same stuff the task is adding for them. I don't know why they'd want to, but they could. So, if a field _could_ be provided by the user, I think it should be in the schema
> 
> I'm not completely sure I buy this argument - it seems to me that being theoretically able to set this already in the RPA is more of a side effect. I'm not even sure if that should be allowed at all - it should say what was actually included in that release, right? So users shouldn't really be able to say that something extra is being released too when it's not. Or am I missing something?

I am referring to adding it to the Release data section, not ReleasePlanAdmission

**Comment by mmalina** [comment]
> > > > > If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json
> > > > 
> > > > 
> > > > This is added during the release pipeline. I thought the schema only covers what's there at the start, but apparently I was wrong. What's the purpose of having releaseNotes.content.images there? It will never be validated, no?
> > > 
> > > 
> > > The releaseNotes stuff is unique. We don't overwrite. We `=+` https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/populate-release-notes-images/populate-release-notes-images.yaml#L111 A user could define the same stuff the task is adding for them. I don't know why they'd want to, but they could. So, if a field _could_ be provided by the user, I think it should be in the schema
> > 
> > 
> > I'm not completely sure I buy this argument - it seems to me that being theoretically able to set this already in the RPA is more of a side effect. I'm not even sure if that should be allowed at all - it should say what was actually included in that release, right? So users shouldn't really be able to say that something extra is being released too when it's not. Or am I missing something?
> 
> I am referring to adding it to the Release data section, not ReleasePlanAdmission

What difference does it make? I understand that if you create a manual Release then you know the snapshot and so you can enter correct data about images in the snapshot into the Release data section, but those would then be duplicated in our task anyway. So not sure what the point would be.

**Comment by johnbieren** [comment]
> > > > > > If you are adding things to releaseNotes the schema needs to be updated https://github.com/konflux-ci/release-service-catalog/blob/development/schema/dataKeys.json
> > > > > 
> > > > > 
> > > > > This is added during the release pipeline. I thought the schema only covers what's there at the start, but apparently I was wrong. What's the purpose of having releaseNotes.content.images there? It will never be validated, no?
> > > > 
> > > > 
> > > > The releaseNotes stuff is unique. We don't overwrite. We `=+` https://github.com/konflux-ci/release-service-catalog/blob/development/tasks/populate-release-notes-images/populate-release-notes-images.yaml#L111 A user could define the same stuff the task is adding for them. I don't know why they'd want to, but they could. So, if a field _could_ be provided by the user, I think it should be in the schema
> > > 
> > > 
> > > I'm not completely sure I buy this argument - it seems to me that being theoretically able to set this already in the RPA is more of a side effect. I'm not even sure if that should be allowed at all - it should say what was actually included in that release, right? So users shouldn't really be able to say that something extra is being released too when it's not. Or am I missing something?
> > 
> > 
> > I am referring to adding it to the Release data section, not ReleasePlanAdmission
> 
> What difference does it make? I understand that if you create a manual Release then you know the snapshot and so you can enter correct data about images in the snapshot into the Release data section, but those would then be duplicated in our task anyway. So not sure what the point would be.

Because you want an image in your advisory that is not part of your snapshot or isn't mapped or something. I don't know. I am just saying it is a possible use case so it should be in the schema

**Comment by jedinym** [comment]
> I replied on the ticket, but why are we adding new fields to the advisories? Shouldn't this be driven by prodsec? Maybe it already is, but I haven't seen it

@johnbieren If I understand correctly, the new fields in the advisory are not the intention here. They are only a side effect of adding additional data for the update-component-sbom task. I will refactor this PR to avoid that issue.

**Comment by johnbieren** [comment]
> > I replied on the ticket, but why are we adding new fields to the advisories? Shouldn't this be driven by prodsec? Maybe it already is, but I haven't seen it
> 
> @johnbieren If I understand correctly, the new fields in the advisory are not the intention here. They are only a side effect of adding additional data for the update-component-sbom task. I will refactor this PR to avoid that issue.

Thank you!

**Comment by jedinym** [comment]
@mmalina @johnbieren I refactored the PR to avoid changing the advisories. It's ready for review, thanks. I'd like to merge this before #627.

File: tasks/update-component-sbom/update-component-sbom.yaml
Comment (mmalina) [inline tasks/update-component-sbom/update-component-sbom.yaml]: This is the updated image that contains https://github.com/konflux-ci/release-service-utils/pull/333 , right? Could you please update also the test reference of the image? We like to keep all references within a task folder in sync.

Comment (mmalina) [reply tasks/update-component-sbom/update-component-sbom.yaml]: Or maybe the other task needs the updated image as well?

File: tasks/populate-release-notes-images/README.md
Comment (mmalina) [inline tasks/populate-release-notes-images/README.md]: Can you reword this to make it clear that it's a path to a file with the data and not the actual data in the result? (Same applies to the change log entry.)

File: tasks/populate-release-notes-images/README.md
Comment (johnbieren) [inline tasks/populate-release-notes-images/README.md]: `s/image data/image data to a file/`

File: tasks/populate-release-notes-images/README.md
Comment (johnbieren) [inline tasks/populate-release-notes-images/README.md]: Let's go to 2.3.0

File: tasks/populate-release-notes-images/populate-release-notes-images.yaml
Comment (johnbieren) [inline tasks/populate-release-notes-images/populate-release-notes-images.yaml]: let's go to 2.3.0

File: tasks/populate-release-notes-images/populate-release-notes-images.yaml
Comment (johnbieren) [inline tasks/populate-release-notes-images/populate-release-notes-images.yaml]: Might be a bit nicer to have a variable for `$(dirname "$(params.dataPath)")/sbom_data.json` instead of writing it twice, but a bit of a nit pick

File: tasks/populate-release-notes-images/populate-release-notes-images.yaml
Comment (johnbieren) [inline tasks/populate-release-notes-images/populate-release-notes-images.yaml]: `s/the result/an sbom data file/`

File: tasks/populate-release-notes-images/populate-release-notes-images.yaml
Comment (johnbieren) [inline tasks/populate-release-notes-images/populate-release-notes-images.yaml]: Does it really need to be in releaseNotes.content.images now that it isn't being combined with the advisory stuff? Seems like just an unnecessary layer of complexity

Comment (jedinym) [reply tasks/populate-release-notes-images/populate-release-notes-images.yaml]: I opened a PR in the utils repo to refactor: https://github.com/konflux-ci/release-service-utils/pull/342.

**Comment by jedinym** [comment]
@johnbieren @mmalina I implemented your requested changes.

**Review by johnbieren — APPROVED** [review]

File: tasks/update-component-sbom/update-component-sbom.yaml
Comment (jedinym) [reply tasks/update-component-sbom/update-component-sbom.yaml]: I updated the tags to use changes from https://github.com/konflux-ci/release-service-utils/pull/342.

Comment (jedinym) [reply tasks/update-component-sbom/update-component-sbom.yaml]: And populate-release-notes does not need that image.

Comment (mmalina) [reply tasks/update-component-sbom/update-component-sbom.yaml]: ok, thanks

**Review by mmalina — APPROVED** [review]

**Comment by mmalina** [comment]
Note that stage Pyxis is down ATM, so e2e will fail right now.

**Comment by jedinym** [comment]
@mmalina Pyxis is up, could you please run the tests?

**Comment by mmalina** [comment]
Let's see if it is enough to start the tests.

**Comment by johnbieren** [comment]
E2E passed and this is approved. It is going to be competing with some other PRs for the rebase thing before merging, but now that we can rebase via UI, I will handle rebasing and merging this today. @wcheang this is ready to be merged, right?

**Comment by jedinym** [comment]
@johnbieren wcheang is on holiday, I took over this PR. It is ready to be merged. 

Just a heads-up, rebasing via the UI will remove signatures from commits.

**Comment by johnbieren** [comment]
> @johnbieren wcheang is on holiday, I took over this PR. It is ready to be merged.
> 
> Just a heads-up, rebasing via the UI will remove signatures from commits.

thanks for the heads up. Yeah, we removed the signed commit requirement because having our contributors rebase 90x with us unable to help was not good :slightly_smiling_face: so now we can rebase for you

**Pull request merged**
